oceanwaves/spectral.py

```diff
@@ -29,14 +31,40 @@ def jonswap(f, Hm0, Tp, gamma=3.3, sigma_low=.07, sigma_high=.09,
         Method to compute alpha (default: yamaguchi)
     normalize : bool
         Normalize resulting spectrum to match ``Hm0``
-    
+
     Returns
     -------
     E : numpy.ndarray
-        Array of shape ``f`` with wave energy densities 
-    
+        Array of shape ``f, Hm0.shape`` with wave energy densities
+
     '''
 
+    # C Stringari - 04/06/2018
+    # check input data types to avoid the following error:
+    # ValueError: Integers to negative integer powers are not allowed.
+
+    # raise an warning if the frequency array starts with zero. if the
+    # user gives an array with zeros, the output will be inf at that
+    # frequency
+    if 0.0 in f:
+        warn("frequency array constains zeros.")
+
+    # get the input dtypes and promote to float 64, if needed
+    f = _check_dtype(f)
+    Hm0 = _check_dtype(Hm0)
+    Tp = _check_dtype(Tp)
+
+    # check shapes of Hm0 and Tp
+    if type(Hm0) == np.ndarray:
+        if Hm0.shape != Tp.shape:
+            raise ValueError("Dimensions of Hm0 and Tp should match.")
+
+    # This is a very naive implementation to deal with array inputs,
+    # but will work if Hm0 and Tp are vectors. I don't know how to expand
+    # the array to N dimensions nicely - C. Stringari
+    if type(Hm0) == np.ndarray:
+            f = np.repeat(f, Hm0.shape[0]).reshape([len(f), Hm0.shape[0]])
```

Comment:
Here we can use the following construct:

```
f = f[:,np.newaxis].repeat(len(Hm0), axis=1)
Hm0 = Hm0[np.newaxis,:].repeat(len(f), axis=0)
Tp= Tp[np.newaxis,:].repeat(len(f), axis=0)
```

to ensure that all three matrices are equal in size.